recipes/z5py/build.sh

```diff
@@ -0,0 +1,67 @@
+# Platform-specific dylib extension
+if [ $(uname) == "Darwin" ]; then
+    export CC=clang
+    export CXX=clang++
+    export DYLIB="dylib"
+else
+    export CC=gcc
+    export CXX=g++
+    export DYLIB="so"
```

Comment:
You can use `${SHLIB_EXT}` for dylib/so https://conda.io/docs/user-guide/tasks/build-packages/environment-variables.html.

Also `$CC` and `$CXX` should select these compiler names correctly upon `{{ compiler('cxx') }}` and `{{ compiler('c') }}` for `requirements/build/` in meta.yaml.